ephyviewer/datasource/epochs.py

```diff
@@ -54,152 +55,219 @@ class WritableEpochSource(InMemoryEpochSource):
     epoch is dict
     { 'time':np.array, 'duration':np.array, 'label':np.array, 'name':' ''}
     """
-    def __init__(self, epoch, possible_labels, color_labels=None):
-        InMemoryEpochSource.__init__(self, all_epochs=[epoch])
+    def __init__(self, epoch=None, possible_labels=[], color_labels=None, channel_name=''):
         
-        #~ self._t_stop = max([ np.max(e['time']+e['duration']) for e in self.all if len(e['time'])>0])
+        self.possible_labels = possible_labels
+        self.channel_name = channel_name
         
-        self.times = self.all[0]['time']
-        self.durations = self.all[0]['duration']
+        if epoch is None:
+            epoch = self.load()
+        
+        InMemoryEpochSource.__init__(self, all_epochs=[epoch])
+        
+        # assign each epoch a fixed, unique integer id
+        self._next_id = 0
+        for chan in self.all:
+            chan['id'] = np.arange(self._next_id, self._next_id + len(chan['time']))
+            self._next_id += len(chan['time'])
         
         assert self.all[0]['time'].dtype.kind=='f'
         assert self.all[0]['duration'].dtype.kind=='f'
-        assert np.all((self.times[:-1]+self.durations[:-1])<=self.times[1:])
```

Comment:
Maybe this could be kept and test or not with an optional karg like **assert_epoch_not_overlap_at_load=True**

Reply:
Sure, we could add that.